lib/Sema/CSSolver.cpp

```diff
@@ -652,6 +652,22 @@ bool ConstraintSystem::tryTypeVariableBindings(
         type = type->reconstituteSugar(/*recursive=*/false);
       }
 
+      // If the type binding comes from the argument conversion, let's
+      // instead of binding collection types directly let's try to
+      // bind using temporary type variables substituted for element
+      // types, that's going to ensure that subtype relationship is
+      // always preserved.
+      if ((binding.BindingSource == ConstraintKind::ArgumentConversion ||
```

Comment:
"else if"?

Reply:
Yes, I forgot to change that from previous version, will do!